internal/buffer/save.go

```diff
@@ -115,28 +115,22 @@ func (b *Buffer) saveToFile(filename string, withSudo bool, autoSave bool) error
 			return
 		}
 
-		// end of line
-		var eol []byte
-		if b.Endings == FFDos {
-			eol = []byte{'\r', '\n'}
-		} else {
-			eol = []byte{'\n'}
+		if e = b.Backup(true); e != nil {
+			return
 		}
 
-		// write lines
-		if fileSize, e = file.Write(b.lines[0].data); e != nil {
+		var readCloser io.ReadCloser
+		readCloser, e = b.GetBackupFile()
```

Comment:
Do we need this hassle? Can't we just write the buffer to the file (the same buffer that we've just written to the backup)? We still have this buffer in memory, right?

Reply:
Sure, we've it still available.
I've been stuck on this backup idea and directly used it for the copy action.
At least it was shorter from code point of view within this path.

But if you insist to use the backup just as a fallback file, then we can do it the other way around.